crypto/pkcs11key/key.go

```diff
@@ -307,8 +474,12 @@ func (ps *Key) openSession() (pkcs11.SessionHandle, error) {
 		// practice it appears to be okay to login to a token multiple times with the same
 		// credentials.
 		if err = ps.module.Login(session, pkcs11.CKU_USER, ps.pin); err != nil {
-			ps.module.CloseSession(session)
-			return session, err
+			if err == pkcs11.Error(pkcs11.CKR_USER_ALREADY_LOGGED_IN) {
```

Comment:
I have a token like that as well: a Yubikey Neo in PIV mode. As the code is currently written, I think it's not safe to instantiate multiple `Key`'s. After each Sign, `Key` will call `Logout`. That means that if you instantiate two `Key`s, one will `Login` sucessfully, and with this code, the second will silently succeed even though the login attempt failed. However, after the first `Key` signs something, the second `Key` will think it's logged in but not be.

Generally I work around this by passing `-cpus 1` to pkcs11key_bench_test when I run it on my Yubikey.

So, my recommendation would be to pull this code out for this change. I'm definitely open to being convinced this can work, but I also think it's best as a separate branch.